File: src/src/expr/primary/pointer.cpp
#include <eval.h>

#include <primary/pointer.hpp>
parserTypes::primary::Pointer::Pointer(Expr& expr)
    : BasePrimary(), pointer(expr) {}

parserTypes::primary::Pointer::~Pointer() {}
void parserTypes::primary::Pointer::eval(parserCore& ctx, int dest) {
  ::eval::Ptr(&ctx, this->pointer, dest);
}

File: parser/expr/value/pointer.py
from ..exprbase import ExprBase
from .value import Value


class Pointer(Value):
    def __init__(self, expr: ExprBase):
        self.expr = expr

    def __str__(self):
        return f'[{self.expr}]'

    def __repr__(self):
        return f'Pointer({self.expr})'
